File: test_auth.js
const http = require('http');

function req(path, headers, label) {
  return new Promise((resolve) => {
    const req = http.request({ hostname: '127.0.0.1', port: 18789, path, method: 'GET', headers, timeout: 5000 }, (res) => {
      let d = '';
      res.on('data', c => d += c);
      res.on('end', () => {
        try {
          resolve({ status: res.statusCode, body: JSON.parse(d) });
        } catch {
          resolve({ status: res.statusCode, body: d });
        }
      });
    });
    req.on('error', e => resolve({ status: 'ERR', body: String(e) }));
    req.on('timeout', () => { req.destroy(); resolve({ status: 'TIMEOUT', body: 'timeout' }); });
    req.end();
  });
}

(async () => {
  const r1 = await req('/api/health', {}, 'NO_AUTH');
  console.log('NO_AUTH:', r1.status, r1.body);

  const r2 = await req('/api/health', { 'X-API-Key': 'wrong' }, 'WRONG_KEY');
  console.log('WRONG_KEY:', r2.status, r2.body);

  const r3 = await req('/api/health', { 'X-API-Key': 'nova_admin_2026' }, 'CORRECT_KEY');
  console.log('CORRECT_KEY:', r3.status, r3.body);

  console.log('ENV:', process.env.DASHBOARD_API_KEY);
  console.log('ENV_DEFAULT:', process.env.DASHBOARD_API_KEY || 'nova_admin_2026');
})();

File: test_auth.py
import os
import httpx

# Test 1 — no key (should be 403)
try:
    r = httpx.get('http://127.0.0.1:18790/api/health')
    print('NO_AUTH:', r.status_code)
except Exception as e:
    print('NO_AUTH_EX:', e)

# Test 2 — wrong key (should be 403)
try:
    r = httpx.get('http://127.0.0.1:18790/api/health', headers={'X-API-Key': 'wrong'})
    print('WRONG_KEY:', r.status_code)
except Exception as e:
    print('WRONG_KEY_EX:', e)

# Test 3 — correct key (should be 200)
# Reads the key from the environment - never hardcode a live credential.
try:
    r = httpx.get('http://127.0.0.1:18790/api/health', headers={'X-API-Key': os.getenv('DASHBOARD_API_KEY', '')})
    print('CORRECT_KEY:', r.status_code, r.json())
except Exception as e:
    print('CORRECT_KEY_EX:', e)

# Test 4 — env var check
import os
print('ENV:', repr(os.getenv('DASHBOARD_API_KEY')))
print('ENV_SET:', bool(os.getenv('DASHBOARD_API_KEY')))
